### PR TITLE
[onert] Remove AUTO from train loss reduction type

### PR DESCRIPTION
**Pull request by zetwhite**
This PR removes AUTO from train loss reduction type.

ONE-DCO-1.0-Signed-off-by: SeungHui Youn <sseung.youn@samsung.com>

related issue : https://github.com/Samsung/ONE/issues/12418

### CODE REVIEW

File: tests/tools/onert_train/src/args.cc [hunk 1]
```diff
@@ -230,9 +230,8 @@ void Args::Initialize(void)
         "1: CATEGORICAL_CROSSENTROPY\n")
     ("loss_reduction_type", po::value<int>()->default_value(0)->notifier([&] (const auto &v) { _loss_reduction_type = v; }),
         "Loss Reduction type\n"
-        "0: AUTO (default)\n"
-        "1: SUM_OVER_BATCH_SIZE\n"
-        "2: SUM\n")
+        "0: SUM_OVER_BATCH_SIZE(default)\n"
+        "1: SUM\n")
```
Comment (zetwhite) [inline tests/tools/onert_train/src/args.cc:234]: (warning) 

Here, `onert_train` argument is modified. 
This could affect developer's local test-bed.

**Review by YongseopKim — DISMISSED** [review]
LGTM 👍 

I'm worried one thing - the enums value match.
- The `NNFW_TRAIN_LOSS_REDUCTION` enum in `runtime/onert/api/include/nnfw_experimental.h` 
- onert_train's arg `loss_reduction_type` 
- would confuse somewhat.

We must not forget to handle them carefully.

(Just IMO, you can ignore.)

**Comment by zetwhite** [comment]
> I'm worried one thing - the enums value match.

Ah, That's right!  I missed that point, thank you a lot for your careful look. 

My minor worry is that It is fine to update `nnfw_experimental.h` values. :thinking: 
Because It could break some client codes that depend on this enum value.

IMO, we could update nnfw_experiment.h for now.  
  * Because there is no client of this training feature except `onert_train`. 
  * Also, I already had a history of updating `nnfw_experimental.h` enum value - https://github.com/Samsung/ONE/pull/12391/files 

So, Let's just update enum value to avoid value miss-match :smile:

**Comment by YongseopKim** [comment]
@zetwhite oh. You don't have to consider my opinion right now. You can take your time enough and consider it later. 😄

**Review by YongseopKim — APPROVED** [review]
LGTM 💯

File: tests/tools/onert_train/src/args.cc (hunk 1 above)
Comment (jyoungyun) [inline tests/tools/onert_train/src/args.cc:234]:
```suggestion
        "1: SUM_OVER_BATCH_SIZE(default)\n"
        "2: SUM\n")
```

File: tests/tools/onert_train/src/args.cc [hunk 2]
```diff
@@ -230,9 +230,8 @@ void Args::Initialize(void)
         "1: CATEGORICAL_CROSSENTROPY\n")
     ("loss_reduction_type", po::value<int>()->default_value(0)->notifier([&] (const auto &v) { _loss_reduction_type = v; }),
```
Comment (jyoungyun) [inline tests/tools/onert_train/src/args.cc:231]:
```suggestion
    ("loss_reduction_type", po::value<int>()->default_value(1)->notifier([&] (const auto &v) { _loss_reduction_type = v; }),
```

File: tests/tools/onert_train/src/onert_train.cc [hunk 3]
```diff
@@ -138,10 +138,8 @@ int main(const int argc, char **argv)
       switch (type)
       {
         case 0:
```
Comment (jyoungyun) [inline tests/tools/onert_train/src/onert_train.cc:140]:
```suggestion
        case 1:
```

File: tests/tools/onert_train/src/onert_train.cc [hunk 4]
```diff
@@ -138,10 +138,8 @@ int main(const int argc, char **argv)
...
-          return NNFW_TRAIN_LOSS_REDUCTION_AUTO;
-        case 1:
           return NNFW_TRAIN_LOSS_REDUCTION_SUM_OVER_BATCH_SIZE;
-        case 2:
+        case 1:
```
Comment (jyoungyun) [inline tests/tools/onert_train/src/onert_train.cc:142]:
```suggestion
        case 2:
```

**Review by jyoungyun — COMMENTED** [review]
How about this one? :)

**Comment by zetwhite** [comment]
> How about this one? :)

No problem :)  

Just for curiosity, May I ask the reason for reverting the args value?
Maybe not change onert_train's behavior? or to avoid value mismatch between nnfw_experiment and onert_train (mentioned here - https://github.com/Samsung/ONE/pull/12419#pullrequestreview-1808847922)

**Comment by jyoungyun** [comment]
> Maybe not change onert_train's behavior? or to avoid value mismatch between nnfw_experiment and onert_train 

Changing the value can satisfy both issues even though it does not match with other parameter's in `onert_train`.

**Review by jyoungyun — APPROVED** [review]
LGTM

**Comment by hseok-oh** [comment]
@zetwhite Please resolve conflicts

**Review by jyoungyun — APPROVED** [review]
LGTM

**Review by YongseopKim — APPROVED** [review]
LGTM 🥇

**Pull request merged**